src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component, Element, Prop, Watch, h } from '@stencil/core';
+
+const VALID_STATES = ['', 'success', 'warning', 'danger'];
+const VALID_SIZES = ['', 'sm', 'md', 'lg', 'xl'];
+
+@Component({
+  tag: 'chi-text-input',
+  styleUrl: 'text-input.scss',
+  scoped: true
+})
+
+export class TextInput {
+  @Element() el: HTMLInputElement;
+  /**
+   * To define state color of Text input
+   */
+  @Prop() state: string;
```

Comment:
Yo can specify valid values like this:
```
@Prop() state: "" | "success" | "warning" | "danger";
```
You can also define a type or an enum and use it globally across whole Chi project, and would be the best approach:
```
export enum CHI_STATES {
  NO_STATE = "", // Empty string would only be acceptable if there is no default value. I.e. default value for size is MD, so there is no need of empty string for size. 
  SUCCESS = "success",
  WARNING = "warning",
  DANGER = "danger"
}
@Prop() state: CHI_STATES;
```
 Please apply to other enum values. @SergioQCL , please validate.

Reply:
Shouldn't the state be a string value since it only accepts one of them?

Reply:
Any of the both solutions I commented accept only those string values. So it is like a string but restricted only to accepted values.